src/commands/deploy.ts

```diff
@@ -180,6 +193,14 @@ export default class Deploy extends Command {
       }
     }
 
+    const extra_params = this.getExtraEnvironmentVariables(flags.parameter);
+    for (const param_name of Object.keys(extra_params)) {
+      if (env_config.getParameters()[param_name] === undefined) {
+        throw new Error(`Parameter ${param_name} not found in env config`);
+      }
+    }
+    env_config.setParameters(extra_params);
```

Comment:
We are still posting the raw_config below. I think its safe to post env_config now - since I deferred the @Transform. 
Make sure to test.